BitmovinConvivaAnalytics/Classes/ContentMetadataBuilder.swift

```diff
@@ -103,11 +105,12 @@ class ContentMetadataBuilder: CustomStringConvertible {
                 contentInfo.merge(additionalStandardTags) { $1 }
             }
         } else {
-            if let duration = self.duration, duration > 0 {
-                if let newDuration = contentInfo[CIS_SSDK_METADATA_DURATION] as? Int {
-                    if newDuration == 0 {
-                        contentInfo[CIS_SSDK_METADATA_DURATION] = duration
-                    }
+            let oldDuration = contentInfo[CIS_SSDK_METADATA_DURATION] as? Int
+            if oldDuration == nil || oldDuration == 0 {
+                if let duration = self.duration, duration > 0 {
+                    contentInfo[CIS_SSDK_METADATA_DURATION] = duration
+                } else {
+                    contentInfo[CIS_SSDK_METADATA_DURATION] = -1
```

Comment:
1. refactored this to make it more clear when we update the value
2. added the handling for unknown/live value and assigning `-1`